question-service/src/routes/questionRoutes.ts

```diff
@@ -0,0 +1,59 @@
+import { Request, Response } from "express";
+import express from "express";
+import Question from "../models/questionModel";
+
+const router = express.Router();
+
+
+// Add a question
+router.post("/addquestion", async (req: Request, res: Response) => {
+  try {
+    const { title, description, difficulty, topics } = req.body;
+    
+    const existing = await Question.findOne({ 
+      title, 
+      difficulty, 
+      topics: { $all: topics, $size: topics.length } 
+    });
+
+    if (existing) {
+      return res.status(400).json({ message: "Duplicate question already exists" });
+    }
```

Comment:
Currently, the 'existing' variable is not null only when it finds a question with the same title, difficulty, and topics. This means that 'exisiting' will be null for a question in the DB that has the same title but different difficulty and topics. 

However since the questions are unique to question title this duplicate question check will fail to catch the problem here, and this issue is caught later on under "Invalid data".

Should check only for title:

```suggestion
    const existing = await Question.findOne({ title });

    if (existing) {
      return res.status(400).json({ message: "A question with this title already exists" });
    }
```